### PR TITLE
cluster: fix the bug change yaml will happen nothing #158

### LINKED ISSUE #158: when the status not update， change replicas and apply the mysql_v1alpha1_cluster.yaml , happen nothing
**Describe the problem**

**To Reproduce**

1. repliacs in yaml is 3, `kubectl apply -f config/samples/mysql_v1alpha1_cluster.yaml` , it can sucess
2. change replicas to 2 then  `kubectl apply -f config/samples/mysql_v1alpha1_cluster.yaml`, it can success
3. Don't wait all pod status change, change replicas to 3 or 5, `kubectl apply -f config/samples/mysql_v1alpha1_cluster.yaml`, it will still in 2 pods , never change to 3 or 5.

**Expected behavior**
```
func (s *StatefulSetSyncer) updatePod(ctx context.Context) error {
	if s.sfs.Status.UpdatedReplicas >= s.sfs.Status.Replicas {
		return nil
	}

	log.Info("statefulSet was changed, run update")

	if s.sfs.Status.ReadyReplicas < s.sfs.Status.Replicas {
		log.Info("can't start/continue 'update': waiting for all replicas are ready")
		return nil  //<---- code here is uncorrect, should return error
	}
```

**Environment:**

- RadonDB MySQL version:

### PR DESCRIPTION
**Pull request by acekingke**
### What type of PR is this?

/bug
### Which issue(s) this PR fixes?

Fixes #158 

### What this PR does?

Summary:

### CODE REVIEW

File: cluster/syncer/statefulset.go [hunk 1]
```diff
@@ -457,7 +457,15 @@ func (s *StatefulSetSyncer) applyNWait(ctx context.Context, pod *corev1.Pod) err
 		if err != nil && !k8serrors.IsNotFound(err) {
 			return false, err
 		}
+		ordinal, err := utils.GetOrdinal(pod.Name)
```
Comment (andyli029) [inline cluster/syncer/statefulset.go:460]: Add space

File: cluster/syncer/statefulset.go [hunk 2]
```diff
@@ -457,7 +457,15 @@ func (s *StatefulSetSyncer) applyNWait(ctx context.Context, pod *corev1.Pod) err
...
+		if err != nil {
+			return false, err
+		}
 
+		if ordinal >= int(*s.Spec.Replicas) {
```
Comment (andyli029) [inline cluster/syncer/statefulset.go:465]: Delete space

**Comment by zhyass** [comment]
LGTM

File: cluster/syncer/statefulset.go [hunk 3]
```diff
@@ -458,6 +458,15 @@ func (s *StatefulSetSyncer) applyNWait(ctx context.Context, pod *corev1.Pod) err
 			return false, err
 		}
 
+		ordinal, err := utils.GetOrdinal(pod.Name)
+		if err != nil {
+			return false, err
+		}
+		if ordinal >= int(*s.Spec.Replicas) {
+			log.Info("retry waiting pod", "pod", pod.Name)
```
Comment (zhyass) [inline cluster/syncer/statefulset.go:466]: The log info need clear describe

**Review by zhyass — CHANGES_REQUESTED** [review]

**Review by zhyass — COMMENTED** [review]
LGTM

**Review by zhyass — APPROVED** [review]

**Review by andyli029 — COMMENTED** [review]
LGTM.

**Pull request merged**